ocs_ci/ocs/resources/storage_cluster.py

```diff
@@ -942,6 +944,67 @@ def add_capacity(osd_size_capacity_requested, add_extra_disk_to_existing_worker=
     return new_storage_devices_sets_count
 
 
+def add_capacity_lso():
+    """
+    Add capacity on LSO cluster.
+
+    In this procedure we need to add the disk before add capacity via UI.
+    Because the UI backend check the pv and available state and base on it
+    change the count param on StorageCluster.
+
+    """
+    from ocs_ci.ocs.cluster import (
+        is_flexible_scaling_enabled,
+        check_ceph_health_after_add_capacity,
+    )
+    from ocs_ci.ocs.ui.helpers_ui import ui_add_capacity_conditions, ui_add_capacity
+
+    node_objs = get_nodes(node_type=constants.WORKER_MACHINE)
+    deviceset_count = get_deviceset_count()
+    if is_flexible_scaling_enabled():
+        add_disk_to_node(node_objs[0])
+        num_available_pv = 1
+    else:
+        add_new_disk_for_vsphere(sc_name=constants.LOCALSTORAGE_SC)
+        num_available_pv = 3
+    localstorage.check_pvs_created(num_pvs_required=num_available_pv)
+    if ui_add_capacity_conditions():
+        try:
+            osd_size = get_osd_size()
+            ui_add_capacity(osd_size)
+        except Exception as e:
+            log.error(
+                f"Add capacity via UI is not applicable and CLI method will be done. The error is {e}"
+            )
+            set_deviceset_count(deviceset_count + 1)
```

Comment:
Here shouldn't it be `set_deviceset_count(deviceset_count + num_available_pv)`?

Reply:
there is a difference between `flexible scaling` and `non flexible scaling`.
On LSO without flexible scaling, the `replica=3`
On LSO with flexible scaling, the `replica=`1`

Reply:
Okay, I understand.